src/parquet/arrow/arrow-reader-writer-test.cc

```diff
@@ -1074,61 +1073,296 @@ class TestNestedSchemaRead : public ::testing::Test {
     writer_->Close();
   }
 
-  void CreateNestedParquet() {
+  void MakeValues(int num_rows) {
+    std::shared_ptr<Array> arr;
+    ASSERT_OK(NullableArray<::arrow::Int32Type>(num_rows, 0, kDefaultSeed, &arr));
+    values_array_ = std::dynamic_pointer_cast<::arrow::Int32Array>(arr);
+  }
+
+  void WriteColumnData(size_t num_rows, int16_t* def_levels,
+      int16_t* rep_levels, int32_t* values) {
+    auto typed_writer = static_cast<TypedColumnWriter<Int32Type>*>(
+      row_group_writer_->NextColumn());
+    typed_writer->WriteBatch(num_rows, def_levels, rep_levels, values);
+  }
+
+  void ValidateArray(std::shared_ptr<Array> array, size_t expected_nulls) {
+    ASSERT_EQ(array->length(), values_array_->length());
+    ASSERT_EQ(array->null_count(), expected_nulls);
+    // Also independently count the nulls
+    auto local_null_count = 0;
+    for (int i = 0; i < array->length(); i++) {
+      if (array->IsNull(i)) {
+        local_null_count++;
+      }
+    }
+    ASSERT_EQ(local_null_count, expected_nulls);
+  }
+
+  void ValidateColumnArray(std::shared_ptr<::arrow::Int32Array> array,
+      size_t expected_nulls) {
+    ValidateArray(array, expected_nulls);
+
+    int j = 0;
+    for (int i = 0; i < values_array_->length(); i++) {
+      if (array->IsNull(i)) {
+        continue;
+      }
+      ASSERT_EQ(array->Value(i), values_array_->Value(j));
+      j++;
+    }
+  }
+
+  // A parquet with a simple nested schema
+  void CreateSimpleNestedParquet(Repetition::type struct_repetition) {
```

Comment:
Right. Basically that is why I added the single type deep schema parquet tests (which uses `CreateMultiLevelNestedParquet`).
I thought the hand-crafted schema test (which uses this `CreateSimpleNestedParquet` function) will become too elaborated if I test that too. 
Do you think that there's an added edge case there to make it worthwhile?